src/lib/grid-list/grid-list.ts

```diff
@@ -110,6 +110,15 @@ export class MdGridList implements OnInit, AfterContentChecked {
 
   /** Creates correct Tile Styler subtype based on rowHeight passed in by user */
   private _setTileStyler(): void {
+    if (this._tileStyler) {
+      // leftover styles must be cleared before setting a new TileStyler
+      this._tiles.forEach((tile, index) => {
+        this._tileStyler.clearRowStyles(tile);
+      });
+      // clear styles set on grid by _layoutTiles()
+      this._setListStyle(['paddingBottom', '0px']);
```

Comment:
This code isn't as clean because it does not defer to the TileStyler implementation to clear the MdGridList styles.

But I wanted to just add a quick fix before adding another method to TileStyler.

I was thinking of changing clearRowStyles to:
`clearTileStyles(tiles: MdGridTile[]):  [string, string]`

which would return the style to clear the grid styling.